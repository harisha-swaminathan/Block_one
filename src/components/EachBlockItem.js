import React from 'react'
import JSONPretty from 'react-json-pretty';

export default class EachBlockItem extends React.Component{
    constructor(){
        super()
        this.state={
            open:false
        }
    }
    onBlockItemClick(){
        this.setState((prevState)=>{
            return {open:!prevState.open}
        })
    }
    
    render(){
        const time=this.props.block.timestamp
        const id=this.props.block.id
        const actions=this.props.actions
       
        return(
            <div className="app">
                <div onClick={()=>{this.onBlockItemClick()}} className="block__item">
                  {/* <span className="block__item-title">Time stamp:</span> */}
                    <p>{time}</p>
                  {/* <span className="block__item-title">Id:</span> */}
                    <p>{id}</p>
                    <p>{`${actions.length}`}</p>
                  {this.state.open && <JSONPretty themeClassName="block__item--open" data={this.props.block}></JSONPretty>}
                </div>
                  
            </div>
      
        )
    }
}