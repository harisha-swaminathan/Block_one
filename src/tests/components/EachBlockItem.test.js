import React from 'react';
import { shallow } from 'enzyme';
import EachBlockItem from '../../components/EachBlockItem'
import {actions, block} from '../fixtures/sampleBlockDetails'


test('renders the component correctly with fixture data', ()=>{
    const wrapper= shallow(<EachBlockItem actions={actions} block={block}/>)
   expect(wrapper.debug()).toMatchSnapshot()
})

test('calls onBlockItemClick method on button click', ()=>{
    const onClickMock = jest.fn();
    EachBlockItem.prototype.onBlockItemClick=onClickMock
    const wrapper = shallow(<EachBlockItem actions={actions} block={block}/>);
    wrapper.find('div').at(1).simulate('click')
    expect(onClickMock).toHaveBeenCalled()
})