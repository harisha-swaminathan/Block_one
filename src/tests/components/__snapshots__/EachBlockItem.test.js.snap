// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component correctly with fixture data 1`] = `
"<div className=\\"app\\">
  <div onClick={[Function: onClick]} className=\\"block__item\\">
    <p />
    <p>
      sampleId123
    </p>
    <p>
      5
    </p>
  </div>
</div>"
`;
